tests/Set/union.php

```diff
@@ -79,3 +91,23 @@ public function testUnionWithSelf(array $initial, array $values, array $expected
     //     $this->assertEquals($initial, $a->toArray());
     // }
 }
+
+class Box implements \Ds\Hashable
+{
+    private $value;
+
+    public function __construct($value)
+    {
+        $this->value = $value;
+    }
+
+    public function equals($obj) : bool
+    {
+        return $this->value === $obj->value;
+    }
+
+    public function hash()
+    {
+        return 0;
```

Comment:
This value doesn't appear to matter. I tried with the same and different values across `Box` instances.